include/onnxruntime/core/providers/cuda/cuda_context.h

```diff
@@ -28,38 +28,70 @@
   cudnnHandle_t cudnn_handle = {};
   cublasHandle_t cublas_handle = {};
   OrtAllocator* deferred_cpu_allocator = {};
+  // below are cuda ep options
+  int16_t device_id = 0;
+  bool has_user_compute_stream = false;
+  size_t gpu_mem_limit = 0;
```

Comment:
>gpu_mem_limit [](http://example.com/codeflow?start=9&length=13)

the memory limit / arena strategy only applied to Cuda EP's bfc arena, I don't see why cstom kernle need it.